### PR TITLE
Grid refinement

### PR DESCRIPTION
**Pull request by McBs**
Hey,

I have implemented a function to refine a fine grid to a course grid.

Mario

### CODE REVIEW

**Comment by McBs** [comment]
oh my mistake!

done.

File: tests/test_util.py [hunk 1]
```diff
@@ -48,15 +48,19 @@ def test_grid_fine_to_couse_2d():
...
+    shear_c_init = lattice.shear_tensor(sim_c.f)
+    shear_c = lattice.shear_tensor(f_c)
+
 
     assert (lattice.u(f_c).numpy() == pytest.approx(u_init.numpy()))
     assert (lattice.rho(f_c).numpy() == pytest.approx(rho_init.numpy()))
     assert (f_c.numpy() == pytest.approx(sim_c.f.numpy()))
+    assert (shear_c_init.numpy() == pytest.approx(shear_c.numpy()))
 
 def test_grid_fine_to_couse_3d():
```
Comment (Olllom) [inline tests/test_util.py:65]: another one

File: tests/test_util.py [hunk 2]
```diff
@@ -33,3 +35,53 @@ def test_torch_gradient_3d(order):
         np.sin(grid[0]) * (-np.cos(grid[1])) * np.sin(grid[2])
     ])
     assert (u0_grad_analytic[0,0,:,0] == pytest.approx(u0_grad[0,0,:,0], rel=2))
+
+def test_grid_fine_to_couse_2d():
```
Comment (Olllom) [inline tests/test_util.py:39]: here's one more

File: tests/test_util.py (hunk 1 above)
Comment (Olllom) [reply tests/test_util.py:65]: (the line below)

**Comment by Olllom** [comment]
Cool, @McBs 
There is one aspect that I overlooked before: the shear moments are probably always zero since you are only looking at a field that has just been initialized (with f=feq). It would be good to test this also with nonzero second moments. One way to go about this would be to initialize the shear moments with the true values (from finite differences). This would be extremely useful to have anyway.

@dominikwilde @McBs 
Do you think we can include the initialization of higher-order moments in this PR? 

One more point:
Now that you are on it, it would be cool to also have the reverse direction (coarse to fine). It requires some kind of interpolation in space though which is not entirely trivial to implement. It's up to you if you want to invest the time since I don't think you need it right at this point. But it will certainly be useful later.

To sum up:
If you are not willing to implement any of this in the present PR, I am happy to merge it as is. But I think that this would be a very good opportunity to add these features. What do you think?

**Comment by McBs** [comment]
done
